airtable/airtable.py

```diff
@@ -1,582 +1,918 @@
+                See :any:`ViewParameter`.
+            sort (``list``, optional): List of fields to sort by.
+                Default order is ascending. See :any:`SortParameter`.
+        Returns:
+            record (``dict``): Deleted Record
+        """
+        record = self.match_in_table(table_name, field_name, field_value, **options)
+        record_url = self.record_table_url(table_name, record["id"])
+        return self._delete(record_url)
+
+    def batch_delete(self, table_name, record_ids):
+        """
+        Calls :any:`delete` repetitively, following set API Rate Limit (5/sec)
+        To change the rate limit set value of ``airtable.API_LIMIT`` to
+        the time in seconds it should sleep before calling the function again.
+        >>> record_ids = ['recwPQIfs4wKPyc9D', 'recwDxIfs3wDPyc3F']
+        >>> airtable.batch_delete('table_name', records_ids)
+        Args:
+            table_name(``str``): Airtable table name
+            records(``list``): Record Ids to delete
+        Returns:
+            records(``list``): list of records deleted
+        """
+        table_delete = partial(self.delete_in_table, table_name)
+        return self._batch_request(table_delete, record_ids)
+
+    def mirror_in_table(self, table_name, records, **options):
+        """
+        Deletes all records on table or view and replaces with records.
+        >>> records = [{'Name': 'John'}, {'Name': 'Marc'}]
+        >>> record = airtable.mirror_in_table('table_name', records)
+        If view options are provided, only records visible on that view will
+        be deleted.
+        >>> record = airtable.mirror_in_table('table_name', records, view='View')
+        ([{'id': 'recwPQIfs4wKPyc9D', ... }], [{'deleted': True, ... }])
+        Args:
+            table_name(``str``): Airtable table name
+            records(``list``): Records to insert
+        Keyword Args:
+            max_records (``int``, optional): The maximum total number of
+                records that will be returned. See :any:`MaxRecordsParameter`
+            view (``str``, optional): The name or ID of a view.
+                See :any:`ViewParameter`.
+        Returns:
+            records (``tuple``): (new_records, deleted_records)
+        """
+
+        all_record_ids = [r["id"] for r in self.get_all(table_name, **options)]
+        deleted_records = self.batch_delete_in_table(table_name, all_record_ids)
+        new_records = self.batch_insert_in_table(table_name, records)
+        return (new_records, deleted_records)
+
+
+class Airtable(AirtableBase):
+
+    def __init__(self, base_key, table_name, api_key=None):
+        """
+        If api_key is not provided, :any:`AirtableAuth` will attempt
+        to use ``os.environ['AIRTABLE_API_KEY']``
+        """
+        super().__init__(base_key, api_key=api_key)
+        self.table_name = table_name
+
+        # the 2 lines below are not really needed but kept just for passing tests
```

Comment:
Can we fix tests instead?

Reply:
yes tests can be easily fixed, but it's more for backward compatibility 
more info [here](https://github.com/gtalarico/airtable-python-wrapper/pull/47/files/8df390a1c2054f0dc35385c5ed71f57953a8adc9#diff-6de8259f9857d7a7fa0a29f9cbb124dcR554)